ratelimits/transaction.go

```diff
@@ -83,7 +84,7 @@ func newFQDNSetBucketKey(name Name, orderNames []string) (string, error) { //nol
 	if err != nil {
 		return "", err
 	}
-	id := fmt.Sprintf("%x", core.HashNames(orderNames))
+	id := fmt.Sprintf("%x", core.HashIdentifiers(identifier.SliceNewDNS(orderNames)))
```

Comment:
Either:
a) everywhere the rate limits package takes in an `orderName(s)`, it should instead take in `orderIdent(s)`; or
b) we should have a way to compute this that doesn't rely on them being identifiers.

As much as we talked about changing HashNames to HashIdentifiers, maybe both stick around and HashIdentifiers just uses HashNames as an inner helper function?